instrumentation-annotations-incubator/src/main/java/io/opentelemetry/instrumentation/annotations/incubator/MetricAttribute.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.annotations.incubator;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation marks that a parameter of a method or constructor annotated with {@link Timed} or
+ * {@link Counted} should be added as an attribute to the instrument.
+ *
+ * <p>Application developers can use this annotation to signal OpenTelemetry auto-instrumentation
+ * that the attribute should be created.
+ *
+ * <p>If you are a library developer, then probably you should NOT use this annotation, because it
+ * is non-functional without the OpenTelemetry auto-instrumentation agent, or some other annotation
+ * processor.
+ */
+@Target(ElementType.PARAMETER)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface MetricAttribute {
+
+  /**
+   * Optional name of the attribute.
+   *
+   * <p>If not specified and the code is compiled using the `{@code -parameters}` argument to
```

Comment:
(I stopped reading after the first line and wondered what it would be if not supplied, I avoiding the <p> will make it flow better)
```suggestion
   * Optional name of the attribute.
   * If not specified and the code is compiled using the `{@code -parameters}` argument to
```

Reply:
This is copied from https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/0cfe778f357fafd36ca505e3f1a12a7a9a2f984e/instrumentation-annotations/src/main/java/io/opentelemetry/instrumentation/annotations/SpanAttribute.java#L29-L33

Reply:
I'd leave this as it is for now since it is copied.